inject/src/main/java/io/micronaut/inject/ast/ClassElement.java

```diff
@@ -236,6 +382,27 @@ default List<PropertyElement> getBeanProperties() {
         return Collections.emptyList();
     }
 
+    /**
+     * Returns the native bean properties (synthetic getters and setters) for this class element..
+     *
+     * @return The bean properties for this class element
+     * @since 4.0.0
+     */
+    default List<PropertyElement> getNativeBeanProperties() {
```

Comment:
Rename to `getSyntheticBeanProperties` since native might be confused with `getNativeType` which returns the underlying type from the underlying AST (Java or Groovy)

Also add `@NonNull`